dysnix/arbitrum/templates/scripts/_init-from-s3.tpl

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env sh
+# shellcheck disable=SC2086,SC3037
+
+set -e
+
+. /scripts/s3-env.sh
+
+process_inputs() {
+  # download even if already initialized
+  if [ "$FORCE_INIT" = "True" ]; then
+    echo "Force init enabled, existing data will be deleted."
+    rm -f "$INITIALIZED_FILE"
+  fi
+  # check for S3 credentials
+  if [ -z "$S3_ENDPOINT_URL" ] || [ -z "$AWS_ACCESS_KEY_ID" ] || [ -z "$AWS_SECRET_ACCESS_KEY" ]; then
+    echo "S3 credentials are not provided, exiting"; exit 1
+  fi
+}
+
+progress() {
+  remote_stats=$("$S5CMD" cat "s3://${STATS_URL}")
+  case $1 in
+  "start")
+    while true; do
+      inodes=$(df -Phi "$DATA_DIR" | tail -n 1 | awk '{print $3}')
+      size=$(df -P -BG "$DATA_DIR" | tail -n 1 | awk '{print $3}')G
+      echo -e "$(date -Iseconds) | SOURCE TOTAL ${remote_stats} | DST USED Inodes:\t${inodes} Size:\t${size}"
+      sleep 2
+    done &
+    progress_pid=$! ;;
+  "stop")
+    kill "$progress_pid"
+    progress_pid=0 ;;
+  "*")
+    echo "Unknown arg" ;;
+  esac
+}
+
+check_lockfile() {
+  if "$S5CMD" cat "s3://${LOCKFILE_URL}" >/dev/null 2>&1; then
+    echo "Found existing lockfile, snapshot might be corrupted. Aborting download.."
+    exit 1
+  fi
+}
+
+# stop all background tasks
+interrupt() {
+  echo "Got interrupt signal, stopping..."
+  for i in "$@"; do kill $i; done
+}
+
+sync() {
+  # check if we are already initialized
+  if [ -f "$INITIALIZED_FILE" ]; then
+    echo "Blockchain already initialized. Exiting..."
+    exit 0
+  fi
+  # s5cmd does not support "true" sync, it does not save object's timestamps
+  # https://github.com/peak/s5cmd/issues/532
+  echo "Cleaning up local data..."
+  rm -rf "${DATA_DIR:?}/"*
```

Comment:
Basically this syntax is an additional check, so if $DATA_DIR is not provided, it will throw an error instead of doing rm -rf /*

Reply:
Ok, WDYT on adding this obvious thing into the commend above?
Do we need an `*` here ?

Reply:
Dependent on files count we may encounter "argument list too long" error, so maybe just deleting and re-creating the folder is a good option. However it can lead to some problems when uid of s3 init container is different from main.

Reply:
My concern with `*` is the expected skip of  "hidden" objects with names with the leading dot. Better way of the cleanup may be a directory re-creation, yes.